File: DSA_Question/valid_anagram/valid_anagram.py
class Solution :
    def isAnagram(self, s:str, t:str)-> bool:
        if len(s)!=len(t):
            return False
        else:
            count_s={}
            count_t = {}

            for i in range(len(s)):
                count_s[s[i]] = 1 + count_s.get(s[i],0)
                count_t[t[i]] = 1 + count_t.get(t[i],0)

            for key in count_s:
                if count_s[key] != count_t[key]:
                    return False
            return True


if __name__ == "__main__":
    objs = Solution()
    return_val =objs.isAnagram("Rupesh","hsepuRsfjadf")
    print(return_val)

File: DSA_Question/valid_anagram/ValidAnagram.java
package valid_anagram;
import java.util.HashMap;
import java.util.Map;

public class ValidAnagram {
    public boolean isAnagram(String s, String t) {
        if (s.length()!= t.length()) {
            return false;
        }

        Map<Character, Integer> countS = new HashMap<>();
        Map<Character, Integer> countT = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            countS.put(s.charAt(i), countS.getOrDefault(s.charAt(i), 0) + 1);
            countT.put(t.charAt(i), countT.getOrDefault(t.charAt(i), 0) + 1);
        }

        return countS.equals(countT);
    }

    public static void main(String[] args) {
        ValidAnagram obj = new ValidAnagram();
        boolean retVal = obj.isAnagram("Rupesh", "hsepuRsfjadf");
        System.out.println(retVal);
    }
}